src/endpoint/cert_worker.js

```diff
@@ -0,0 +1,44 @@
+/* Copyright (C) 2023 NooBaa */
+'use strict';
+
+const dbg = require('../util/debug_module')(__filename);
+const config = require('../../config.js');
+const ssl_utils = require('../util/ssl_utils');
+
+/**
+ * Updates s3 https server(s) when certificate files are updated.
+ * Note ODF rotates certificates every 13 months.
+ * Some more details in operator's doc/ssl-dns-routing.md -
+ * https://github.com/noobaa/noobaa-operator/blob/master/doc/ssl-dns-routing.md
+ */
+class CertWorker {
+
+    /**
+     * @param {{
+     *   name: string;
+     *   https_server: https.Server;
+     *   sts_server: https.Server;
+     * }} params
+     */
+    constructor({ name, https_server, sts_server }) {
+        this.name = name;
+        this.https_server = https_server;
+        this.sts_server = sts_server;
+    }
+
+    async run_batch() {
+        dbg.log2('certificate bg worker: running check for certificate update');
+        if (await ssl_utils.update_certs_from_disk()) {
+            //certificate were updated. give new cert to https servers.
+            dbg.log0('certificate bg worker: updating certs');
+            const ssl_cert = await ssl_utils.get_ssl_certificate('S3');
+            const ssl_options = { ...ssl_cert, honorCipherOrder: true };
+            this.https_server.setSecureContext(ssl_options);
+            this.sts_server.setSecureContext(ssl_options);
+        }
+        //use same delay as core
+        return config.CLUSTER_HB_INTERVAL;
```

Comment:
I think we should maybe use a different one than HeartBeat. I think once a day should be enough no? maybe with error handling retry it more frequently(1 min?) CERT_WORKER_DELAY and CERT_WORKER_ERROR_DELAY.

Reply:
That's what Ohad did initially.
Also if user deletes the old certificates manually (because they were hacked and can't be used anymore) you don't want to wait a day for new certificate to kick in.

Reply:
OK, so it can be lower (half an hour?|)... just think the 2 shouldn't be connected and it's confusing. also maybe it should also have an enabled/disabled config in case it's self-gen and we are not interested in it at all. I think the user can always delete the pod in case of an emergency.